src/System.Device.Gpio/System/Device/ComponentInformation.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+#pragma warning disable CS1591
+namespace System.Device
+{
+    public record ComponentInformation
```

Comment:
It should be a reference type, since it anyway contains references to other instances. The reason it is not readonly has to do with creating it in derived classes. I  need to look at that again.

Reply:
Done, the record is now read-only. Fortunately, records can still be easily "updated" by using a copy and the `with` keyword.

Reply:
> records can still be easily "updated" by using a copy and the with keyword
Oh well, you get a copy of the object with one or more mutated fields. The original object did not change at all.